rules/TypeDeclaration/Rector/ClassMethod/AddParamTypeBasedOnPHPUnitDataProviderRector.php

```diff
@@ -133,7 +133,7 @@ public function refactor(Node $node): ?Node
 
             $dataProviderNodes = $this->resolveDataProviderNodes($classMethod);
             if ($dataProviderNodes->isEmpty()) {
-                return null;
+                continue;
```

Comment:
This fixes an unrelated problem, previously if a method did not have a data provider we would return from this function so that it wouldn't process other methods. Instead we need to continue so that it can process them

Reply:
sounds like this would need a unit test which fails without the change